### PR TITLE
Add GIF support

### PR DESCRIPTION
**Pull request by ok-nick**
## Changes in this pull request
Adds GIF support.

### Related Issues
* Completes #476 
* Partially completes #394

### WIP Priorities
- [x] Parse entire GIF past pre-image-data block extensions
  - [x] Enables box hashing
  - [x] Completes XMP support (need to search pre and post block extensions)
- [x] Block extension parser should be incremental (via iterator)
  - [x] It should also iterate blocks past pre-image-data as described above
- [x] ~~Data hash location for XMP block~~
- [x] Add unit tests
  - [x] Composed manifest refs
  - [x] ~~Asset patches~~
  - [x] Data hash locations
  - [x] Box hash locations
  - [x] Remote refs

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/asset_handlers/gif_io.rs [hunk 1]
```diff
@@ -0,0 +1,1453 @@
...
+        let data_written = gif_io.read_cai(&mut output_stream1)?;
+        assert_eq!(data_written, random_bytes);
+
+        let mut output_stream2 = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 3));
+        let random_bytes = [3, 2, 1, 2, 3];
+        gif_io.write_cai(&mut output_stream1, &mut output_stream2, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream2)?;
+        assert_eq!(data_written, random_bytes);
+
```
Comment (mauricefisher64) [inline sdk/src/asset_handlers/gif_io.rs:1303]: An interesting sanity check here would be to see if "stream" matches SAMPLE1.  The contents should be identical after these transformation.

**Comment by mauricefisher64** [comment]
Looks good.  Just address the build issues and I can approve.

**Comment by ok-nick** [comment]
@mauricefisher64 Build issue is caused by an update in nightly toolchain, fixed in #498

**Review by mauricefisher64 — APPROVED** [review]
👍

**Pull request merged**